js/src/behaviors/tabs.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * Tab control for assessment lens
+ * @param container - Container of tab control component
+ */
+/**
+ * Tabs behaviors
+ */
+export function initTabs() {
+	initTabNavClickListeners();
+	initTabControlClickListeners();
+
+	// update when tabs into the view
+	const tabContainers = Array.from(document.querySelectorAll('tab-container')) as HTMLElement[];
+	for (const tc of tabContainers) {
+		updateTabNav(tc);
+	}
+}
+
+function initTabNavClickListeners() {
+	window.addEventListener('click', (event: Event) => {
+		const target =
+			event.target instanceof Element &&
+			(event.target.closest('[data-tab-nav]') as HTMLButtonElement);
+
+		if (!target) {
+			return;
+		}
+
+		event.preventDefault();
+		const container = event.target.closest('tab-container') as HTMLElement;
+		const tabs = Array.from(container.querySelectorAll('.tab-control')) as HTMLButtonElement[];
+
+		// get current activated tab
+		const currentActiveButton = container.querySelector(
+			'[aria-selected="true"]'
+		) as HTMLButtonElement;
+		const index = parseInt(currentActiveButton?.dataset.tabControl as string);
+
+		// update tab
+		if (index > 1 && target.dataset.tabNav === 'previous') {
+			updateTabState(container, currentActiveButton, index - 1);
+			updateTabPanel(container, index, index - 1);
+		} else if (index < tabs.length && target.dataset.tabNav === 'next') {
+			updateTabState(container, currentActiveButton, index + 1);
+			updateTabPanel(container, index, index + 1);
+		}
+		updateTabNav(container);
+	});
+}
+
+function initTabControlClickListeners() {
+	window.addEventListener('click', (event: Event) => {
+		const target =
+			event.target instanceof Element &&
+			(event.target.closest('.tab-control') as HTMLButtonElement);
+
+		if (!target) {
+			return;
+		}
+
+		event.preventDefault();
+		const container = event.target.closest('tab-container') as HTMLElement;
+		updateTabNav(container);
+	});
+	window.addEventListener('keydown', (event: KeyboardEvent) => {
+		if (event.key !== 'ArrowLeft' && event.key !== 'ArrowRight') {
+			return;
+		}
+
+		const target =
+			event.target instanceof Element &&
+			(event.target.closest('.tab-control') as HTMLButtonElement);
+
+		if (!target) {
+			return;
+		}
+
+		event.preventDefault();
+		const container = event.target.closest('tab-container') as HTMLElement;
+		updateTabNav(container);
+	});
+}
+
+function updateTabNav(container: HTMLElement = document.body) {
+	const tabs = Array.from(container.querySelectorAll('.tab-control')) as HTMLButtonElement[];
```

Comment:
data- attrs or web components for js functionality ditto for this function

Reply:
updated